### PR TITLE
Fixing precious problems and Adding functions for streaming

### PR DESCRIPTION
**Pull request by pinghaoluo**

### CODE REVIEW

**Comment by pinghaoluo** [comment]
@allenyhu @barath  I meet a weird bugs. When I try register or doing all other command, sometimes it  does't work. Like I try 10 times and 2 times it works. I tried to print testing message in serviceLayer and KV store. I found sometimes there is no connection between serviceLayer client and server. And sometimes there is no connection between server and KV store. Is there something wrong with my virtual machine?

**Review by barath — APPROVED** [review]

File: src/cli/chirp.cc [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #include <gflags/gflags.h>
 
 #include "cli_parser.h"
-
+using namespace std;
```
Comment (allenyhu) [inline src/cli/chirp.cc:6]: i don't think you're supposed to use global namespaces

**Review by allenyhu — APPROVED** [review]

**Comment by allenyhu** [comment]
@pinghaoluo I've never experienced those errors. Are you working off the same vm that you coded phases 1&2 in? When I tried to use my old vm for your project it wouldn't work either. Another reason that could be happening is that you're trying to issue commands before starting the KVServer and the ServiceServer

File: src/cli/chirp.cc (hunk 1 above)
Comment (pinghaoluo) [reply src/cli/chirp.cc:6]: Fixed, I used it for debugging and forget to delete

**Pull request merged**